### PR TITLE
introducing first time user to our app

### LINKED ISSUE #1381: adding a new feature i.e. introducing  first time user to  the app
**Summary:** 

when a new user uses the app ,app will show the instructions and features ,so that user can get  easily familiarized with app

**Expected behaviour:** 
The app will start and guide the first time user and show them all the features of the app .

### PR DESCRIPTION
**Pull request by neeraj0403**
Fixes #1381 

Please Add Screenshots If there are any UI changes.
ezgif com-video-to-gif (8)

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by jawidMuhammadi** [comment]
@neeraj0403 Please make different branches for each issue you solve and raise PR accordingly.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java [hunk 1]
```diff
@@ -64,14 +66,15 @@
     View mNavigationHeader;
     SwitchCompat userStatusToggle;
     private Menu menu;
+    Context context;
```
Comment (jawidMuhammadi) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java:69]: Why we need context object inside activity, can't we use `this`?

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/SplashScreenActivity.java [hunk 2]
```diff
@@ -36,5 +42,44 @@ protected void onCreate(Bundle savedInstanceState) {
...
+
+                    runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            startActivity(INTENT);
+                        }
+                    });
+
+                    //  Make a new preferences editor
+                    SharedPreferences.Editor E = PREF.edit();
```
Comment (jawidMuhammadi) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/SplashScreenActivity.java:73]: Do not do all the Preference related operations inside the activity. Use predefined helper methods from the `PrefManager` class in the `utils` folder.  Give the variables a descriptive name using **Camel Case** naming style

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/SplashScreenActivity.java [hunk 3]
```diff
@@ -36,5 +42,44 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    private  void startappintro(final Context context) {
+        mthread = new Thread(new Runnable() {
+                @Override
+            public void run() {
+                //  Initialize SharedPreferences
+                SharedPreferences PREF = PreferenceManager
+                        .getDefaultSharedPreferences(context);
+
+                //  Create a new boolean and preference and set it to true
+                misFirstStart = PREF.getBoolean("firstStart", true);
```
Comment (jawidMuhammadi) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/SplashScreenActivity.java:57]: Please, avoid using hardcoded string, it is not a good practice.

File: mifosng-android/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -200,6 +200,7 @@
     <string name="add_payment_detail">Add Payment Detail</string>
     <string name="generate_new">Generate New</string>
     <string name="currency">Currency</string>
+    <string name="firststart_splash_screen">firststart</string>
```
Comment (jawidMuhammadi) [inline mifosng-android/src/main/res/values/strings.xml:203]: Add this string in `constans.java` file or add the `translatable=false`  attribute to it.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/SplashScreenActivity.java [hunk 5]
```diff
@@ -50,11 +51,10 @@ private  void startappintro(final Context context) {
                 @Override
             public void run() {
                 //  Initialize SharedPreferences
-                SharedPreferences PREF = PreferenceManager
-                        .getDefaultSharedPreferences(context);
+                mprefManager = PrefManager.getPreferences();
 
                 //  Create a new boolean and preference and set it to true
-                misFirstStart = PREF.getBoolean("firstStart", true);
+                misFirstStart = mprefManager.getBoolean("firstStart", true);
```
Comment (jawidMuhammadi) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/SplashScreenActivity.java:57]: We have predefined methods named `putBoolean`  and `getBoolean` in `PrefManager.java` for saving and retrieving boolean from preferences. Please those methods.

**Comment by AbhilashG97** [comment]
Hello @neeraj0403, 

Please squash your commits into a single commit. Also, it would be great if you could re-open the issue for this pull request and discuss the implementation of this feature with the community.

I'm closing this pull request for now. Please re-open this pull request after you have made the suggested changes.

Thank you for your contribution.

**Pull request closed without merging**